packages/nameguard-js/src/nameguard-js.test.ts

```diff
@@ -0,0 +1,70 @@
+import { describe, it, expect } from "vitest";
+import { createPublicClient, http } from "viem";
+import { mainnet } from "viem/chains";
+import { createClient } from "./nameguard-js";
+
+const PROVIDER_URI_MAINNET = process.env.PROVIDER_URI_MAINNET;
+
+describe("NameGuardJS", () => {
+  it("should compute secure primary name", async () => {
+    const publicClient = createPublicClient({
+      chain: mainnet,
+      transport: http(PROVIDER_URI_MAINNET),
+    });
+
+    const localNameguard = createClient({
+      // not a real endpoint, will error if used
+      endpoint: "http://localhost:1234",
+      publicClient,
+    });
+
+    const data = await localNameguard.getSecurePrimaryName(
+      "0xb8c2C29ee19D8307cb7255e1Cd9CbDE883A267d5",
+    );
+
+    expect(data.display_name).toBe("nick.eth");
+  });
+
+  it("should use the API for requests with a different network", () => {
```

Comment:
I think we can remove this test case. Please see other related comments. Suggest we enforce a single network for everything related to a `NameGuard` instance for its entire lifetime.